jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js

```diff
@@ -42,15 +48,32 @@ YUI.add('deployment-payment', function() {
     },
 
     componentWillMount: function() {
+      this._getUser();
+      this._getCountries();
+    },
+
+    componentWillUnmount: function() {
+      this.xhrs.forEach((xhr) => {
+        xhr && xhr.abort && xhr.abort();
+      });
+    },
+
+    /**
+      Get the payment details for the user.
+
+      @method _getUser
+    */
+    _getUser: function() {
       this.setState({loading: true}, () => {
         const xhr = this.props.getUser(this.props.username, (error, user) => {
-          if (error) {
+          if (error && error !== 'not found') {
+            const message = 'Could not load user info';
             this.props.addNotification({
-              title: 'Could not load user info',
-              message: `Could not load user info: ${error}`,
+              title: message,
+              message: `${message}: ${error}`,
               level: 'error'
             });
-            console.error('Could not load user info', error);
+            console.error(message, error);
             return;
           }
           this.setState({loading: false}, () => {
```

Comment:
It's slightly more complicated than that, when the user finishes loading we may not always be displaying the form that needs the countries. Maybe we should display a loading spinner when loading the form, but in my next branch I'm going to be splitting all the form parts out into separate components so they can be shared with the account page, so maybe the loading could be implemented then.